client/src/index.jsx

```diff
@@ -0,0 +1,206 @@
+import ShareButtons from './components/ShareButtons.jsx'
+import MainPhoto from './components/MainPhoto.jsx'
+import QuantityMenu from './components/QuantityMenu.jsx'
+import StarRatingComponent from 'react-star-rating-component';
+import axios from 'axios';
+import primeLogo from '../dist/primelogo.png'
+ 
+class App extends React.Component {
+    constructor() {
+      super();
+      this.state = {
+        answers: 0,
+        description: '',
+        isHovered: false,
+        modalOpened: false,
+        showingModal: false,
+        photos: [],
+        prime: false,
+        mainphoto: '',
+        rating: 4.5,
+        selectedoption: '',
+        selectValue: '',
+        title: ''
+      }
+      this.getPhotos = this.getPhotos.bind(this);
+      this.hoverExit = this.hoverExit.bind(this);
+      this.hoverHandler = this.hoverHandler.bind(this);
+      this.handleModal = this.handleModal.bind(this);
+      this.hoverOptionHandler = this.hoverOptionHandler.bind(this);
+      this.handleQuantityChange = this.handleQuantityChange.bind(this);
+      this.showModal = this.showModal.bind(this);
+      this.hideModal = this.hideModal.bind(this);
+    }
+  
+    renderView() {
+    }
+
+    getMainProducts() {
+      axios.get('/api/mainProduct').then((data) => {
+        let mainProductData = data.data;
+        // console.log(mainProductData, 'line 48')
+        let test = Math.floor(Math.random() * 10);
+        mainProductData = mainProductData.filter(element => element.product_id === test)
+        console.log(mainProductData, 'line 51')
+        this.setState({
+          answers: mainProductData[0].answers,
+          company: mainProductData[0].company,
+          description: mainProductData[0].description,
+          price: mainProductData[0].price,
+          prime: mainProductData[0].prime,
+          ratings: mainProductData[0].ratings,
+          title: mainProductData[0].title
+        })
+      })
+    }
+  
+    getPhotos() {
+      axios.get('/api/photos').then((data) => {
+        let photourldata = data.data;
+        console.log(photourldata, 'line 52')
+        let photourlarray = [];
+        // makes a random number from 0 to 9
+        let test = Math.floor(Math.random() * 10);
+        photourldata = photourldata.filter(element => element.product_id === test)
+        console.log(photourldata, "logging data from api photos call");
+        // TODO: access string inside object
+        photourldata.forEach(element => {photourlarray.push(element.photo_url)
+        console.log(photourlarray, 'logging data from photourldata line 26')
+        });
+        // TODO: refactor filter method of getting correct urls
+          this.setState({
+            photos: photourlarray
+          });
+      })
+    }
+
+    handleModal(e) {
+      console.log('we out here in handleModal, line 62')
+      this.setState({
+        isHovered: true,
+        modalOpened: true
+      })
+    }
+
+    showModal () {
```

Comment:
might be clearer to rename this toggle modal since it's both hiding and showing it!